[Review] Add option to only playback streams in standard definition (SD)

linqcan opened the pull request:
Needs review.

This option was added as a hack for ATV2 which has problems playing streams encoded using h264 with the avc1.77.30 profile.
The plugin will only playback streams with a maximum resolution of 1024x576 if the option is enabled.

This one includes pull 33 (!).

linqcan committed:
Add option to force non avc1.77.30 streams

This commit forces the use of the 1024x576 or 704x396 streams instead of the 1280x720, if no bandwidth limit is set in XBMC.
It also divides the settings in to two sections; General and Advanced.

nilzen commented:
My bad, merged the other PR before I read this one...

nilzen commented:
I prefer something more general than atv2hack for the method name and also the setting. Perhaps "Exclude avc1.77.30 streams"

linqcan commented:
This comment has been updated!  (I messed up)

Rewrote it according to your suggestions and changed it completely.
For some reason XBMC decided that the lowest quality stream of the avc1.66.30 streams was the one to use.
Put when you do the trick kokangit suggests it sets the 1024x576 stream, strange...

So, I rewrote it to ALWAYS use the 1024x576 (or 704x396) stream instead. I had to look for two different versions since some programs (news,sports) only are produced in 704x396 as highest resolution. The "algorithm" is weak and depends on the order of the streams in the .m3u8 file. But as long as they order them by resolution ascending (which seems logical when looking at the HLS standard) we should be alright.
Please try and review.
Also, I hid the option on an "Advanced" page since it makes no sense for Average Joe.

So what have I done here?
Well, this patch is sort of the same thing as limiting the bandwidth in XBMC settings, BUT it only affects SVTPlay since we are actually manipulating the playlist file from SVTPlay. So I guess it could make sense to have this feature if you don't want to change the bandwidth limit in the system settings when switching between video plugins.

Before you merge (if you decide to do that), let me know and I will squash all commits on this branch into one to prettify the git log.

nilzen commented:
I'm unable to merge this, I guess you have to rebase it.

linqcan commented:
Yep, haven't had time to fix that yet. Will do!

linqcan commented:
Should be OK now. Worked on ATV2.
